src/test/java/hudson/plugins/sonar/action/SonarCacheActionTest.java

```diff
@@ -90,18 +93,34 @@ public void testResolveUsingInstallationUrl() {
 
   @Test
   public void testResponseCached() {
-    ProjectInformation mocked = createProj(now(), "success");
-    SonarAnalysisAction analysis = createAnalysis("serverUrl", "projUrl", "taskId");
+    ProjectInformation mocked1 = createProj(now(), "success");
+    ProjectInformation mocked2 = createProj(now(), "error");
+    SonarAnalysisAction analysis1 = createAnalysis("serverUrl", "projUrl", "taskId1");
+    SonarAnalysisAction analysis2 = createAnalysis("serverUrl", "projUrl", "taskId2");
     Run<?, ?> run = mock(Run.class);
 
-    when(resolver.resolve("serverUrl", "projUrl", "taskId", "inst", run)).thenReturn(mocked);
-    List<ProjectInformation> projs = cache.get(resolver, 0, Collections.singletonList(analysis), run);
-    List<ProjectInformation> projs2 = cache.get(resolver, 0, Collections.singletonList(analysis), run);
+    when(resolver.resolve(analysis1.getServerUrl(), analysis1.getUrl(), Objects.requireNonNull(analysis1.getCeTaskId()), analysis1.getInstallationName(), run)).thenReturn(mocked1);
+    when(resolver.resolve(analysis2.getServerUrl(), analysis2.getUrl(), Objects.requireNonNull(analysis2.getCeTaskId()), analysis2.getInstallationName(), run)).thenReturn(mocked2);
+
+    List<ProjectInformation> projs = cache.get(resolver, 0, Collections.singletonList(analysis1), run);
+    // Calling it again in quick succession, even with more analyses, should still return the cached value.
+    List<ProjectInformation> projs2 = cache.get(resolver, 0, Arrays.asList(analysis1, analysis2), run);
 
-    assertThat(projs).isEqualTo(projs2);
```

Comment:
This assertion was misleading. It's always the same object, regardless of the fact if you hit the cache or not. It's the size that can tell whether the cache was hit or not.